src/main/java/com/swp/board/domain/BoardService.java

```diff
@@ -0,0 +1,131 @@
+package com.swp.board.domain;
+
+import com.swp.auth.dto.JwtUserDetails;
+import com.swp.board.dto.BoardDeleteRequestDto;
+import com.swp.board.dto.BoardCreateRequestDto;
+import com.swp.board.dto.BoardResponseDto;
+import com.swp.board.dto.BoardUpdateRequestDto;
+import com.swp.board.exception.BoardForbiddenException;
+import com.swp.board.exception.BoardNotFoundException;
+import com.swp.study.Exception.StudyNotFoundException;
+import com.swp.study.domain.Study;
+import com.swp.study.domain.StudyRepository;
+import com.swp.user.domain.Role;
+import com.swp.user.domain.User;
+import com.swp.user.domain.UserRepository;
+import com.swp.user.exception.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RequiredArgsConstructor
+@Service
+public class BoardService {
+
+    private final BoardRepository boardRepository;
+    private final UserRepository userRepository;
+    private final StudyRepository studyRepository;
+
+    @Transactional
+    public void createBoard(BoardCreateRequestDto boardCreateRequestDto){
+        JwtUserDetails userDetails = (JwtUserDetails) SecurityContextHolder.getContext()
+                .getAuthentication()
+                .getPrincipal();
```

Comment:
우리 userDetails는 컨트롤러에서 받고, Service로 넘겨주는건 어떨까요?
서비스에는 서비스 로직**만** 들어있으면 좋을것 같아요.

```java
void someControllerMethod(){
    JwtUserDetails userDetails = (JwtUserDetails) SecurityContextHolder.getContext()
        .getAuthentication()
        .getPrincipal();
    boardService.createBoard(userDetails, boardCreateRequestDto);
}
```

```suggestion
    @Transactional
    public void createBoard(JwtUserDetails userDetails, BoardCreateRequestDto boardCreateRequestDto){
        
```

Reply:
네 그렇게 하는 게 좋을 것 같네요. Relationship에 있는 거는 이거 merge 하고 나중에 따로 할게요

Reply:
이건 제가 조금 더 고민을 해봤는데, 아예 **DTO**를 사용해서 처리하는 편이 나을 것 같네요.
지금 상태로는 나중에 인증 방식을 바꾸게 된다면 (~~그렇지는 않지만~~) 서비스를 전부다 건드려야 합니다.
나중에 같이 리팩터링 하면서 (~~할 시간이 있을지는...~~ ) 인증된 유저를 가져오는 방식부터 바꿔야 할 것으로 보입니다.
일단은 통일만 해놓는 것으로 합시다.

Reply:
넵